build/pkgs/ecm/spkg-configure.m4

```diff
@@ -18,8 +19,9 @@ SAGE_SPKG_CONFIGURE([ecm], [
                         AC_SEARCH_LIBS([ecm_factor], [ecm], [], [sage_spkg_install_ecm=yes])
                     ])
                 ])
-                AC_PATH_PROG([ECMBIN], [ecm])
+                AC_PATH_PROGS([ECMBIN], [ecm gmp-ecm])
                 if test x$ECMBIN != x; then
+                    SAGE_ECM_BIN=`basename $ECMBIN`
```

Comment:
why basename

Reply:
pretty sure that's a `-z` test in disguise, too

Reply:
Because `sage.interfaces.ecm` takes `ecm` or `gmp-ecm` as the data it needs to call the executable, not the full path. So it assumes it's on PATH, I'm just keeping it like this here.

Reply:
Maybe the PATH was only set at `configure` time?

Reply:
Same argument applies for the previous design. How is this relevant to this PR?